tools/docvalidator/main.go

```diff
@@ -0,0 +1,165 @@
+// Package main provides a utility to process markdown documentation files,
+// extract code examples, execute them, and verify their outputs.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+	"time"
+)
+
+// Example represents an individual code example extracted from the markdown files.
+// It contains the function name, the file it was extracted from, the code snippet, and the expected output.
+type Example struct {
+	FuncName string
+	File     string
+	Code     string
+	Expected string
+	Skipped  bool
+}
+
+var logLevel string
+
+// init initializes the logging level from command-line flags.
+// It sets the default logging level to "info", but can be overridden by
+// specifying a different level (debug, info, warn, error) using the -log flag.
+func init() {
+	flag.StringVar(&logLevel, "log", "info", "Set the logging level (debug, info, warn, error)")
+}
+
+// main is the entry point of the application that processes markdown documentation files
+// to extract, execute, and verify code examples. It performs the following steps:
+// 1. Parses command-line flags to set the logging level.
+// 2. Sets the log level based on the flag value and forces the timezone to UTC.
+// 3. Discovers markdown files in the 'docs/registries' directory, filtering out specific files.
+// 4. Iterates over each file, extracting code examples and processing them.
+// 5. For each example, logs the processing status and counts errors.
+// 6. Exits with an error code if any examples fail processing, or with success if all pass.
+func main() {
+	// Parse command-line flags
+	flag.Parse()
+
+	// Create an instance of the real file system
+	fs := &OSFileSystem{}
+
+	// Call the run function
+	if err := run(fs); err != nil {
+		// If an error occurs, log it and exit with code 1
+		slog.Error("An error occurred during validation", "error", err)
+		os.Exit(1)
+	}
+	// Exit successfully
+	os.Exit(0)
+}
+
+// run contains the main logic of the program.
+// It processes markdown files, extracts code examples, executes them, and verifies their outputs.
+// This function returns an error if any step fails, making it easier to test.
+func run(fs FileSystem) error {
+	// Set the log level based on the flag
+	if err := setLogLevel(logLevel); err != nil {
+		return err
+	}
+
+	// Force timezone to UTC (for date parsing)
+	time.Local = time.UTC
```

Comment:
This sounds like a terrible hack, especially because you don't restore it when you no longer need it

What code require this?

It should be more explicit.

Also, if your code requires this, it means you should have a variable in your templating system to handle the timezone

Reply:
I think this is trying to solve https://github.com/go-sprout/sprout/issues/81

Reply:
Indeed

But the tests should not force the Go time.Local, the code should be able to work without forcing everything. And if something is forced, it has to be on the package level, not the system one.

Reply:
Illegal take a look at the issue.

The current approach is wrong to me

Reply:
I opened
- #94 

Please remove everything about `time.Local` override out of the PR about template examples

Reply:
Gotcha! Thanks @ccoVeille